airbyte-cdk/java/airbyte-cdk/dependencies/src/main/kotlin/io/airbyte/commons/features/EnvVariableFeatureFlags.kt

```diff
@@ -29,32 +32,45 @@ class EnvVariableFeatureFlags : FeatureFlags {
         return getEnvOrDefault(FIELD_SELECTION_WORKSPACES, "") { arg: String -> arg }
     }
 
-    override fun strictComparisonNormalizationWorkspaces(): String? {
+    override fun strictComparisonNormalizationWorkspaces(): String {
         return getEnvOrDefault(STRICT_COMPARISON_NORMALIZATION_WORKSPACES, "") { arg: String ->
             arg
         }
     }
 
-    override fun strictComparisonNormalizationTag(): String? {
+    override fun strictComparisonNormalizationTag(): String {
         return getEnvOrDefault(STRICT_COMPARISON_NORMALIZATION_TAG, "strict_comparison2") {
             arg: String ->
             arg
         }
     }
 
-    override fun deploymentMode(): String? {
+    override fun deploymentMode(): String {
         return getEnvOrDefault(DEPLOYMENT_MODE, "") { arg: String -> arg }
     }
 
+    override fun logInvalidJsonDeserialization(): Boolean {
+        return getEnvOrDefault(LOG_INVALID_JSON_DESERIALIZATION, false) { s: String -> s.toBoolean() }
+    }
+
     // TODO: refactor in order to use the same method than the ones in EnvConfigs.java
-    fun <T> getEnvOrDefault(key: String?, defaultValue: T, parser: Function<String, T>): T {
-        val value = System.getenv(key)
-        if (value != null && !value.isEmpty()) {
-            return parser.apply(value)
-        } else {
-            log.debug { "Using default value for environment variable $key: '$defaultValue'" }
-            return defaultValue
+    // These methods currently have different behavior (info vs debug log; the
+    // ability to mask secret values in the log message).
+    // This implementation also checks System.getProperty(), whereas EnvConfigs
+    // only checks System.getenv().
+    fun <T> getEnvOrDefault(key: String, defaultValue: T, parser: Function<String, T>): T {
+        val envVal = System.getenv(key)
+        if (!envVal.isNullOrEmpty()) {
+            return parser.apply(envVal)
         }
+
+        val propVal = System.getProperty(key)
```

Comment:
any risk with adding the lookup into Sytem.properties?

Reply:
afaik this is fine - unless there happened to be a preexisting system property clashing with a feature flag name, this will be a noop